rust/nasl-interpreter/src/assign.rs

```
@@ -0,0 +1,290 @@
+                let register = self.registrat.last_mut();
+                register.add_named(key, ContextType::Value(NaslValue::Array(arr)));
+                result
+            }
+        };
+        Ok(result)
+    }
+
+    #[inline(always)]
+    fn dynamic_return(
+        &mut self,
+        order: &AssignOrder,
+        key: &str,
+        lookup: Option<NaslValue>,
+        result: impl Fn(&NaslValue, &NaslValue) -> NaslValue,
+    ) -> InterpretResult {
+        match order {
+            AssignOrder::AssignReturn => self.store_return(key, lookup, &NaslValue::Null, result),
+            AssignOrder::ReturnAssign => {
+                let left = self.named_value(key)?;
+                let result = match lookup {
+                    None => {
+                        let result = result(&left, &NaslValue::Null);
+                        let register = self.registrat.last_mut();
+                        register.add_named(key, ContextType::Value(result));
+                        left
+                    }
+                    Some(idx) => {
+                        let (idx, mut arr) = prepare_array(&idx, left);
+                        let orig = arr[idx].clone();
+                        let result = result(&orig, &NaslValue::Null);
+                        arr[idx] = result;
+                        let register = self.registrat.last_mut();
+                        register.add_named(key, ContextType::Value(NaslValue::Array(arr)));
+                        orig
+                    }
+                };
+                Ok(result)
+            }
+        }
+    }
+}
+
+impl<'a> AssignExtension for Interpreter<'a> {
+    fn assign(
+        &mut self,
+        category: TokenCategory,
+        order: AssignOrder,
+        left: Statement,
+        right: Statement,
+    ) -> InterpretResult {
+        let (key, lookup) = {
+            match left {
+                Variable(token) | Array(token, None) => (&self.code[Range::from(token)], None),
```

comment edge case reasoning for Array(token, None)

---

verified via openvas-nasl, doesn't exist and should throw an error.